normandy/base/middleware.py

```diff
@@ -117,3 +122,150 @@ def process_request(self, request):
                 }
             )
         return response
+
+
+class OIDCEndpointRequestError(Exception):
+    """Happens when the server-to-server communication with the OIDC endpoint
+    succeeds but the OIDC endpoints responds with a status code > 200 and less
+    than 500."""
+
+
+class OIDCAccessTokenAuthorizationMiddleware:
```

Comment:
I'm actually ok with the use of the word authorization here, because it is dealing specifically with the authorization HTTP header.